src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Registry.hs

```diff
@@ -29,35 +29,35 @@ import Data.Map.NonEmpty (NEMap)
 import Data.Map.NonEmpty qualified as NEM
 import Swarm.Game.Location (Location)
 import Swarm.Game.Scenario.Topography.Placement (StructureName)
-import Swarm.Game.Scenario.Topography.Structure qualified as Structure
 import Swarm.Game.Scenario.Topography.Structure.Recognition.Type
+import Swarm.Game.Scenario.Topography.Structure.Type qualified as Structure
 import Swarm.Game.Universe (Cosmic)
 import Swarm.Util (binTuples, deleteKeys)
 
 -- | The authoritative source of which built structures currently exist.
-data FoundRegistry = FoundRegistry
-  { _foundByName :: Map StructureName (NEMap (Cosmic Location) StructureWithGrid)
-  , _foundByLocation :: Map (Cosmic Location) FoundStructure
+data FoundRegistry b a = FoundRegistry
```

Comment:
Again, I think we should add a comment here explaining what the type parameters `b` and `a` generally represent.